internal/versions/versions.go

```diff
@@ -153,9 +383,140 @@ func (h *handler) GetReleaseImage(ctx context.Context, openshiftVersion, cpuArch
+	}
+
+	if err := h.validateReleaseImageCPUArchitecture(cpuArchitecture); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// GetReleaseImage returns the release image that matches the specified openshift version, CPU architecture.
+// Whether a release matches is mainly defined by the openshift version argument given:
+// If openshift version is given as major.minor.patch with/without pre-release, only exact match will be returned.
+// If openshift version is given as major.minor, the latest of this minor will be retrieved
+// If openshift version is given as major, the latest of this major will be retrieved
+func (h *handler) GetReleaseImage(
+	ctx context.Context,
+	openshiftVersion,
+	cpuArchitecture,
+	pullSecret string,
+) (*models.ReleaseImage, error) {
+	errorMessageTemplate := "The requested release image for version (%s) and CPU architecture (%s) isn't specified in release images list"
+
+	if err := h.validateReleaseImageVersionAndArchitecture(openshiftVersion, cpuArchitecture); err != nil {
+		return nil, err
+	}
+
+	versionSegmentsLength, err := common.GetVersionSegmentsLength(openshiftVersion)
+	if err != nil {
+		h.log.WithError(err).
+			Debugf("error uccurred while trying to get the number of segments of version: %s", openshiftVersion)
+		return nil, err
+	}
+
+	multiTargetRelease := strings.HasSuffix(openshiftVersion, "-multi")
+	if multiTargetRelease {
+		cpuArchitecture = common.MultiCPUArchitecture
+	}
+
+	// First, try to get the release image from the configuraion
+	image, err := h.getReleaseImageFromConfig(
+		openshiftVersion,
+		cpuArchitecture,
+		*versionSegmentsLength,
+		multiTargetRelease,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	if image != nil {
+		return image, nil
+	}
+
+	// The image doesn't exist in the configuration. Lets try to find it in the db releases
+	releaseImage, err := h.getReleaseImageFromDB(openshiftVersion, cpuArchitecture, *versionSegmentsLength)
+	if err != nil {
+		return nil, err
+	}
+
+	if releaseImage != nil {
+		return releaseImage, nil
+	}
+
+	// If still no match found, fallback to major.minor and try to find the latest release matching
+	majorMinorVersion, err := common.GetMajorMinorVersion(openshiftVersion)
+	if err != nil {
+		h.log.WithError(err).Debugf("error occurred while trying to convert %s", openshiftVersion)
+		return nil, err
+	}
+	// Keep the multi suffix if specified
+	if strings.HasSuffix(openshiftVersion, "-multi") {
+		majorMinorVersion = swag.String(swag.StringValue(majorMinorVersion) + "-multi")
+	}
+
+	// First, from DB
+	releaseImage, err = h.getReleaseImageFromDB(*majorMinorVersion, cpuArchitecture, 2)
+	if err != nil {
+		return nil, err
+	}
+
+	if releaseImage != nil {
+		return releaseImage, nil
+	}
+
+	// Then if not found, from configuration
+	image, err = h.getReleaseImageFromConfig(*majorMinorVersion, cpuArchitecture, 2, multiTargetRelease)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
We should have a clear priority here regardless if we are looking for X.Y.Z or just X.Y.

Right now we're prioritizing the config when we're looking for and exact version but DB comes first when we're looking for latest.